pkg/monitor/monitor/util.go

```diff
@@ -406,7 +406,13 @@ func getMonitorDMInitContainer(monitor *v1alpha1.TidbMonitor, dc *v1alpha1.DMClu
 }
 
 func getMonitorPrometheusContainer(monitor *v1alpha1.TidbMonitor, tc *v1alpha1.TidbCluster) core.Container {
-	commands := []string{"sed 's/$NAMESPACE/'\"$NAMESPACE\"'/g;s/$POD_NAME/'\"$POD_NAME\"'/g' /etc/prometheus/config/prometheus.yml > /etc/prometheus/config_out/prometheus.yml && /bin/prometheus --web.enable-admin-api --web.enable-lifecycle --config.file=/etc/prometheus/config_out/prometheus.yml --storage.tsdb.path=/data/prometheus " + fmt.Sprintf("--storage.tsdb.retention=%dd", monitor.Spec.Prometheus.ReserveDays)}
+	var retention string
+	if monitor.Spec.Prometheus.RetentionTime != nil {
+		retention = *monitor.Spec.Prometheus.RetentionTime
+	} else {
+		retention = fmt.Sprintf("%dd", monitor.Spec.Prometheus.ReserveDays)
```

Comment:
What's the behavior if `monitor.Spec.Prometheus.ReserveDays` is not set? Does `0d` means that the data will be preserved forever?

Reply:
if monitor.Spec.Prometheus.ReserveDays not set, it means `0d` yes, `0d` means default `15d`( if storage.tsdb.retention.size is 0)